vips/govips.go

```diff
@@ -86,6 +86,7 @@ func Startup(config *Config) {
 	// Initialize govips logging handler and verbosity filter to historical default
 	if !currentLoggingOverridden {
 		LoggingSettings(nil, LogLevelInfo)
+		currentLoggingOverridden = false
```

Comment:
Isn't this already false?

Reply:
oh I see, nevermind. Ugh, this is a little messy, but OK

Reply:
Sorry 😄 You get what you pay for 😉

Reply:
The null logging handler is the same technique Libvips uses in C to shut down logging. Not sure if your method works just as well but wanted to be safe. However it seems like the macOS tests are still failing so I guess it wasn't this?

Reply:
Hmm, I was able to repro locally (by making some changes) and then this fixed it. Let me see

Reply:
I think it's the non-deterministic concurrency of the tests. Also the tests never call Shutdown explicitly so, it's not to be expected that the tests will pass in every environment. By default parallelism is set to MAXPROCS, which will vary obviously

Reply:
Unfortunately, we can't call shutdown in tests because libvips does not support shutdown/startup in the same process. Cleanest way I could imagine is to call to manually call our tests as subtests from a single test and shutdown there.

Reply:
Oh man. And Go doesn't have atexit either. I guess we could set MAXPROCS to 1, but is test order execution guaranteed any way? And the macOS CI builds take like 15 minutes already. I'd really like to at least get the CI builds working as they're quite useful...

Reply:
Well, I think my suggestion may be what we're left with.

Or there's something useful in here: https://golang.org/pkg/os/signal/

Reply:
I don't think a program exit can raise a signal, it's only the other way around. I consider this issue purely a CI issue, as you said users should call Shutdown(). Also, this is just interim, as there's a bug also open for the Go runtime for macOS and I hope they fix that someday.

So the simple option is to just disable the logging handler for the unit tests when they're running in Github CI. I'll just write a patch for that.

Reply:
That seems reasonable.

Reply:
Done. Thanks for the help David.